### DISCUSSION TITLE
Await is only valid in async function

### DISCUSSION THREAD

**Discussion by Wamzel** [denysdovhan/wtfjs#184] [WTF Questions] [answered]

**Comment by pinage404** [comment, marked as answer]
`await` is syntax sugar around `Promise`

---

There is two cases to use `await` : 
1. outside of any function / at top level
1. in a function

---

For the first case, [the specification](https://github.com/tc39/proposal-top-level-await) is currently in progress and [will come in the future](https://caniuse.com/mdn-javascript_operators_await_top_level)

---

For the second case, a function that call a `Promise` must : 
* give a `callback` that will be executed after the `Promise` finished the asynchronous task
* let the `Promise` do side-effect
* return a `Promise`

If you care about the result of the `Promise` and want to wait before executing others instructions, that means that this function will wait for an asynchronous task, and be `async`hronous as well, that why the `async` keyword is required

**Comment by denysdovhan** [comment]
This is more like a question. A top-level `await` is coming.
